core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -143,12 +144,24 @@ public FSDataOutputStream append(Path path, int bufferSize, Progressable progres
 
   @Override
   public void close() throws IOException {
+    if (isClosed) {
+      return;
+    }
+
     // super.close should be called first before releasing the resources in this instance, as the
     // super class may invoke other methods in this class. For example,
     // org.apache.hadoop.fs.FileSystem.close may check the existence of certain temp files before
     // closing
     super.close();
     mFileSystem.close();
+    isClosed = true;
+  }
+
+  @Override
+  protected void finalize() throws Throwable {
```

Comment:
The `hadoop-azure` module also uses the finalize method to handle this situation. https://issues.apache.org/jira/browse/HADOOP-10840
Any better suggestions are welcome.